adapters/repos/db/vector/hnsw/delete.go

```diff
@@ -579,9 +579,9 @@ func (h *hnsw) removeTombstonesAndNodes(deleteList helpers.AllowList, breakClean
 
 		h.resetLock.Lock()
 		if !breakCleanUpTombstonedNodes() {
-			h.Lock()
+			h.shardedNodeLocks[id%NodeLockStripe].RLock()
```

Comment:
This lock change broke unit tests (and don't think it should be a `RLock` in this case. I'll revert.